synapse/lib/cell.py

```diff
@@ -1787,11 +1790,9 @@ def _fireActiveCoro(self, iden, cdef):
         func = cdef.get('func')
 
         async def wrap():
-            while not self.isfini:
+            while not self.isfini and self.isActiveCoro(iden):
```

Comment:
tidy up while we're here... the `await asyncio.sleep(1)` below would be better as `await self.waitfini(1)`